packages/react-native-codegen/src/parsers/parsers-primitives.js

```diff
@@ -193,16 +193,36 @@ function emitPromise(
   typeAnnotation: $FlowFixMe,
   parser: Parser,
   nullable: boolean,
+  types: TypeDeclarationMap,
+  aliasMap: {...NativeModuleAliasMap},
+  tryParse: ParserErrorCapturer,
+  cxxOnly: boolean,
+  translateTypeAnnotation: $FlowFixMe,
 ): Nullable<NativeModulePromiseTypeAnnotation> {
   assertGenericTypeAnnotationHasExactlyOneTypeParameter(
     hasteModuleName,
     typeAnnotation,
     parser,
   );
 
-  return wrapNullable(nullable, {
-    type: 'PromiseTypeAnnotation',
-  });
+  const elementType = typeAnnotation.typeParameters.params[0];
+  if (elementType.type === 'ExistsTypeAnnotation') {
```

Comment:
What is `ExistsTypeAnnotation` and why this is excluded?

Reply:
Is the `ExistsTypeAnnotation` is emitted with the same type from both the Flow and the TS parser or does it exists a `TSExistsTypeAnnotation`?

Reply:
It means `*` in `Promise<*>`. In flow it means something similar to `Promise<$FlowFixMe>`, but it is not possible to deduct it in `react-native-codegen`. So the only way is to keep the behavior unchanged in this case.